processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/PartitionContext.java

```diff
@@ -92,6 +93,14 @@ private static SubPartitions createSubPartitions(PartitionScope scope, Processor
     PartitionContext(PartitionScope scope,
                      Processors<?> processors,
                      SubPartitions subPartitions) {
+        this(scope, processors, subPartitions, System::currentTimeMillis);
+    }
+
+    // visible for testing
+    PartitionContext(PartitionScope scope,
+                     Processors<?> processors,
+                     SubPartitions subPartitions,
+                     LongSupplier timestampSupplier) {
```

Comment:
I think we're using `Clock` for such purpose most recently? (to make the intention clear through type)